From df7caafdd74940fb857e17ac35bd7dcbd562b367 Mon Sep 17 00:00:00 2001
From: Uri Mashiach <uri.mashiach@compulab.co.il>
Date: Wed, 16 Sep 2020 16:19:44 +0300
Subject: [PATCH 98/99] iot-gate-imx8: fix UART1 mode GPIO

The operation mode of UART1 (RS232/RS485) is determined by the value of
a GPIO. The GPIO number is updated with the polarity of the GPIO.

Verify that the GPIO number is not modified during the value update
operation.

Signed-off-by: Uri Mashiach <uri.mashiach@compulab.co.il>
---
 board/compulab/iot-gate-imx8/iot-gate-imx8.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/board/compulab/iot-gate-imx8/iot-gate-imx8.c b/board/compulab/iot-gate-imx8/iot-gate-imx8.c
index c6174daab3..1e764c9f29 100644
--- a/board/compulab/iot-gate-imx8/iot-gate-imx8.c
+++ b/board/compulab/iot-gate-imx8/iot-gate-imx8.c
@@ -307,6 +307,10 @@ static void iot_gate_imx8_update_uart1_node(void *blob)
 		return;
 
 	/* Update node regulator-uart1-mode */
+	prop_data_gpio[0]=
+		fdt_getprop_u32_default(blob, IOT_GATE_IMX8_DTB_PATH_UART1_MODE,
+					"gpio", 0);
+	prop_data_gpio[0] = cpu_to_fdt32(prop_data_gpio[0]);
 	do_fixup_by_path(blob, IOT_GATE_IMX8_DTB_PATH_UART1_MODE, "gpio",
 			 prop_data_gpio, sizeof(prop_data_gpio), 0);
 	do_fixup_by_path(blob, IOT_GATE_IMX8_DTB_PATH_UART1_MODE, "enable-active-low",
-- 
2.11.0

